Feature/5285 clonage connecteur metier

laurianemas opened the pull request:
+ deletion

laurianemas committed:
[Evo] #5285 feat: connector cloning function added

laurianemas committed:
[Evo] #5285 feat: duplicate error handled

laurianemas committed:
[Evo] #5285 feat: connector deletion added

bobeal commented on src/main/kotlin/org/ozwillo/dcimporter/service/ConnectorsService.kt:
```diff
@@ -81,6 +81,27 @@ class ConnectorsService(private val businessAppConfigurationRepository: Business
+                        baseUrl = existingConnector.baseUrl,
+                        organizationSiret = siret,
+                        instanceId = existingConnector.instanceId,
+                        login = existingConnector.login,
```
we should avoid copying credential info to enforce editors to choose fresh ones each time they declare a new connector

laurianemas replied:
I'm not sure to understand here. I thought that the cloning logic consists of duplicate the same infos for a new organization. If we ask for new credential infos what is the difference with a classic update form then ?

bobeal replied:
it is not an update, since you create a new connector from some existing info.

but I don't want editors to use the same credential info for **all** their connectors (if one leaks, they all leak).

laurianemas replied:
Aaah ok, I understood. I changed it, now users have to enter their own credential infos in the cloning form (or let them empty if connector don't need it). Application name and application base url are still taken from the existing connector to avoid errors.

bobeal commented on src/main/kotlin/org/ozwillo/dcimporter/service/ConnectorsService.kt:
```diff
@@ -109,7 +130,20 @@ class ConnectorsService(private val businessAppConfigurationRepository: Business
+        val fallback: Mono<BusinessAppConfiguration> =
+                Mono.error(EmptyException("Connector not found for id: \"$id\""))
+
+        return businessAppConfigurationRepository.findById(id)
```
* why loading something you want to delete ?
* why returning something that is deleted ?

laurianemas replied:
About the second point : now it return an empty Mono
For the first one : Without making a findById request and check it I really dont know how to handle a request that will have an incorrect id as parameter.

bobeal replied:
yes ok (and it will be updated when integrating the security part)

bobeal commented on src/main/kotlin/org/ozwillo/dcimporter/service/ConnectorsService.kt:
```diff
@@ -81,6 +81,27 @@ class ConnectorsService(private val businessAppConfigurationRepository: Business
             }
     }
 
+    fun clone(id: String, siret: String): Mono<HttpStatus>{
```
a service should not return a HTTP status

laurianemas replied:
Now it return a Mono<BusinessAppConfiguration> or a Mono.error(ConflictException).

nabdali commented on src/main/resources/public/src/router.js:
```diff
@@ -31,6 +32,11 @@ export default new Router({
             component: ConnectorsManagement
         },
+        {
+            path: '/connectors/:id/:appName/clone',
```
This page can be used for creation, modification and cloning

laurianemas replied:
We need to discuss about it. If so however I suppose that the path should be "/connectors/:id" only.

nabdali commented on src/main/resources/public/src/views/ConnectorsManagement.vue:
```diff
@@ -41,8 +41,14 @@
-                            <td>++</td>
-                            <td>X</td>
+                            <td>
+                                <router-link :to="{ name: 'clone', params: { id: connector.id, appName: connector.applicationName, siret: connector.organizationSiret }}" title="clone connector to an other organization">
```
It can be a button

bobeal replied:
semantically, it's more a link than a button :)

nabdali commented on src/main/resources/public/src/views/ConnectorsManagement.vue:
```diff
@@ -149,19 +155,35 @@ export default {
         },
         getConnectors (siret, appName){
```
```suggestion
        getConnectors (siret, application){
```

laurianemas replied:
Same as below

nabdali commented on src/main/resources/public/src/views/ConnectorsManagement.vue:
```diff
@@ -149,19 +155,35 @@ export default {
         getConnectors (siret, appName){
             this.connectors = []
-            axios.get('/configuration/connectors?siret=' + siret + '&application=' + appName)
+            axios.get('/configuration/connectors', {params: {siret: siret, application: appName}})
```
```suggestion
            axios.get('/configuration/connectors', {params: {siret, application}})
```

laurianemas replied:
I applied it in my last commits chain

nabdali commented on src/main/resources/public/src/views/ConnectorsManagement.vue:
```diff
@@ -149,19 +155,35 @@ export default {
+        deleteConnector (id){
+            axios.delete(`/configuration/connectors/${id}`)
+            .then(() => {
+                this.$router.go()
```
Is better to update the data than force reloading the page

laurianemas replied:
True, corrected

nabdali commented on src/main/resources/public/src/views/CloneConnector.vue:
```diff
@@ -0,0 +1,119 @@
+            })
+            .catch(e => {
+                this.errors.push(e)
+                this.duplicateErrorMessage = ` ${e}: Connector already exists for application ${this.selectedConnector.applicationName} 
```
For the moment we don't manage error message

laurianemas replied:
Removed in last commit. But now if the connector already exist it will only be visible in the browser console as a request resulting in HttpStatus 409

laurianemas committed:
fix: delete connectors by id return empty Mono

In ConnectorsService, method delete

issue #5285

laurianemas committed:
fix: ConnectorService return HttpStatus no more

Fix of create function that now return Mono<BusinessAppConfiguration> or Mono.error(ConflictException) in case of connector already existing for given siret and application name
Conflict exception is catched by onErrorResume that trigger HttpStatus.CONFLICT in ConnectorHandler

fix issue #5285

laurianemas committed:
fix: typo on getConnectors methods from ConnectorsManagement.vue

fix issue #5285

laurianemas committed:
fix: DeleteConnector refresh connectors list instead of reloading page

fix issue #5285

laurianemas committed:
fix: Conflict error handling removed from CloneConnector.vue

fix issue #5285

laurianemas committed:
fix: Add a more complete form for connector cloning

Now the connector object is no more entirely duplicated with only a change on the organization.
Only the application name and base url are duplicated, others credential infos need to be indicated in the corresponding inputs

fix issue #5285

laurianemas committed:
fix: Deletion of remaining CSS and data using in deleted conflict exception handling (commit 9162cb6)
